### ISSUE TITLE
Maven Enunciate is hung for forever

### ISSUE THREAD

**Issue by barkhakeni** [stoicflame/enunciate#93]
I have pom.xml entry as below
<plugin>
   <groupId>org.codehaus.enunciate</groupId>
   <artifactId>maven-enunciate-plugin</artifactId>
   <version>1.29</version>
   <configuration>
      <configFile>enunciate.xml</configFile>
   </configuration>
   <executions>
      <execution>
         <goals>
            <goal>docs</goal>
         </goals>
      </execution>
   </executions>
   <dependencies>
      <dependency>
         <groupId>com.sun</groupId>
         <artifactId>tools</artifactId>
         <version>${java.version}</version>
         <scope>system</scope>
         <systemPath>${java.home}/../lib/tools.jar</systemPath>
      </dependency>
   </dependencies>
</plugin>

My enunciate.xml looks like this

<enunciate label="slib" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
	xsi:noNamespaceSchemaLocation="http://enunciate.codehaus.org/schemas/enunciate-1.27.xsd">	
	<api-import class="test.services.*" />
	<api-import class="test.model.*" />
	<api-import class="test.*" />

	<services>
		<rest defaultRestSubcontext="/test/api" />
	</services>
	<modules>
		<docs css="test.css" title="Test REST API" includeExampleJson="true" includeExampleXml="false" copyright=""></docs>
		<amf disabled="true" />
		<c disabled="true" />
		<obj-c disabled="true" />
		<csharp disabled="true" />
		<cxf disabled="false" />
		<gwt disabled="true" />
		<jaxws-ri disabled="true" />
		<java-client disabled="true" />
		<php disabled="true" />
		<ruby disabled="true" />
		<xml disabled="false">
			<schema namespace="http://www.example.com/v1/System" file="doesnotexist.xsd"/>
		</xml>
		<jersey disable="false"/>	
	</modules>
</enunciate>
When i try to run maven install, it is hung forever after this step
[INFO] initializing enunciate.
[INFO] invoking enunciate:generate step...

warning: The apt tool and its associated API are planned to be
removed in the next major JDK release.  These features have been
superseded by javac and the standardized annotation processing API,
javax.annotation.processing and javax.lang.model.  Users are
recommended to migrate to the annotation processing features of
javac; see the javac man page for more information.
[WARNING] Validation result has warnings.

C:\Users\kenib\kenib-test-sandbox\src\main\java\test\api\TestRestAPI.java:614: warning: [jersey] JAX-RS resource method is designed to catch all requests.
    public JAXBList<ProfileResponse> getProfileForCategory(@PathParam("category") String category, ComplianceModelData filter)
                                     ^
C:\Users\kenib\kenib-test-sandbox\src\main\java\test\api\TestRestAPI.java:1191: warning: [jersey] JAX-RS resource method is designed to catch all requests.
    public JAXBList<TestResponse> setDefault(@PathParam("category") String category, @PathParam("id") String id,
                                        ^

Any ideas? I do have some Json annotations for my model classes. Do i need to do anything?

**Comment by stoicflame** [comment]
Looks like you found a bug.

Unfortunately, I'm going to [need a thread dump](https://helpx.adobe.com/experience-manager/kb/TakeThreadDump.html).

**Comment by barkhakeni** [comment]
I ran maven install through eclipse. I am not sure how to generate a thread dump. Could you please help me?

**Comment by barkhakeni** [comment]
Just realized you have sent a link on how to do it. Will send soon

**Comment by barkhakeni** [comment]
Below is the thread dump

2015-07-16 09:57:24
Full thread dump Java HotSpot(TM) Client VM (24.80-b07 mixed mode, sharing):

"Service Thread" daemon prio=6 tid=0x003dfc00 nid=0x1e68 runnable [0x00000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread0" daemon prio=10 tid=0x003e9800 nid=0x1abc waiting on condition [0x00000000]
   java.lang.Thread.State: RUNNABLE

"Attach Listener" daemon prio=10 tid=0x003df400 nid=0x209c waiting on condition [0x00000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" daemon prio=10 tid=0x003df000 nid=0x1d80 runnable [0x00000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" daemon prio=8 tid=0x00374000 nid=0x1228 in Object.wait() [0x0132f000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x29b50288> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:135)
	- locked <0x29b50288> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:151)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

"Reference Handler" daemon prio=10 tid=0x0036ec00 nid=0x18cc in Object.wait() [0x00e6f000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x29b50318> (a java.lang.ref.Reference$Lock)
	at java.lang.Object.wait(Object.java:503)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:133)
	- locked <0x29b50318> (a java.lang.ref.Reference$Lock)

"main" prio=6 tid=0x00e7c800 nid=0x2048 runnable [0x00efc000]
   java.lang.Thread.State: RUNNABLE
	at java.util.regex.Pattern$Curly.match0(Pattern.java:4158)
	at java.util.regex.Pattern$Curly.match(Pattern.java:4132)
	at java.util.regex.Pattern$GroupTail.match(Pattern.java:4615)
	at java.util.regex.Pattern$Curly.match0(Pattern.java:4170)
	at java.util.regex.Pattern$Curly.match(Pattern.java:4132)
	at java.util.regex.Pattern$GroupHead.match(Pattern.java:4556)
	at java.util.regex.Pattern$BmpCharProperty.match(Pattern.java:3715)
	at java.util.regex.Pattern$Start.match(Pattern.java:3408)
	at java.util.regex.Matcher.search(Matcher.java:1199)
	at java.util.regex.Matcher.find(Matcher.java:592)
	at net.sf.jelly.apt.decorations.JavaDoc.handleAllTags(JavaDoc.java:154)
	at net.sf.jelly.apt.decorations.JavaDoc.init(JavaDoc.java:104)
	at net.sf.jelly.apt.decorations.JavaDoc.<init>(JavaDoc.java:54)
	at net.sf.jelly.apt.decorations.declaration.DecoratedDeclaration.constructJavaDoc(DecoratedDeclaration.java:59)
	at net.sf.jelly.apt.decorations.declaration.DecoratedDeclaration.<init>(DecoratedDeclaration.java:55)
	at net.sf.jelly.apt.decorations.declaration.DecoratedMemberDeclaration.<init>(DecoratedMemberDeclaration.java:29)
	at net.sf.jelly.apt.decorations.declaration.DecoratedTypeDeclaration.<init>(DecoratedTypeDeclaration.java:41)
	at net.sf.jelly.apt.decorations.declaration.DecoratedClassDeclaration.<init>(DecoratedClassDeclaration.java:36)
	at net.sf.jelly.apt.decorations.DeclarationDecorator.visitClassDeclaration(DeclarationDecorator.java:197)
	at com.sun.tools.apt.mirror.declaration.ClassDeclarationImpl.accept(ClassDeclarationImpl.java:113)
	at net.sf.jelly.apt.decorations.DeclarationDecorator.decorate(DeclarationDecorator.java:72)
	at net.sf.jelly.apt.decorations.type.DecoratedClassType.getDeclaration(DecoratedClassType.java:38)
	at net.sf.jelly.apt.decorations.type.DecoratedClassType.getDeclaration(DecoratedClassType.java:27)
	at net.sf.jelly.apt.decorations.type.DecoratedDeclaredType.isInstanceOf(DecoratedDeclaredType.java:66)
	at net.sf.jelly.apt.decorations.type.DecoratedClassType.isInstanceOf(DecoratedClassType.java:50)
	at net.sf.jelly.apt.decorations.type.DecoratedTypeMirror.isCollection(DecoratedTypeMirror.java:73)
	at org.codehaus.enunciate.contract.jaxb.Accessor.isCollectionType(Accessor.java:312)
	at org.codehaus.enunciate.contract.jaxb.Element.isWrapped(Element.java:413)
	at org.codehaus.enunciate.contract.jaxb.Element.getRef(Element.java:163)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:309)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:353)
	at org.codehaus.enunciate.config.SchemaInfo.addReferencedNamespaces(SchemaInfo.java:314)
	at org.codehaus.enunciate.config.SchemaInfo.getReferencedNamespaces(SchemaInfo.java:238)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at freemarker.ext.beans.BeansWrapper.invokeMethod(BeansWrapper.java:866)
	at freemarker.ext.beans.BeanModel.invokeThroughDescriptor(BeanModel.java:277)
	at freemarker.ext.beans.BeanModel.get(BeanModel.java:184)
	at org.codehaus.enunciate.modules.xml.SchemaInfoModel.get(SchemaInfoModel.java:49)
	at freemarker.core.Dot._getAsTemplateModel(Dot.java:76)
	at freemarker.core.Expression.getAsTemplateModel(Expression.java:89)
	at freemarker.core.IteratorBlock.accept(IteratorBlock.java:94)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.MixedContent.accept(MixedContent.java:92)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.Macro$Context.runMacro(Macro.java:172)
	at freemarker.core.Environment.visit(Environment.java:614)
	at freemarker.core.UnifiedCall.accept(UnifiedCall.java:106)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.Environment.visit(Environment.java:310)
	at freemarker.core.CompressedBlock.accept(CompressedBlock.java:73)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.Environment.visit(Environment.java:310)
	at freemarker.core.UnifiedCall.accept(UnifiedCall.java:130)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:79)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.Environment.visit(Environment.java:310)
	at freemarker.core.UnifiedCall.accept(UnifiedCall.java:130)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.MixedContent.accept(MixedContent.java:92)
	at freemarker.core.Environment.visit(Environment.java:221)
	at freemarker.core.Environment.process(Environment.java:199)
	at freemarker.template.Template.process(Template.java:259)
	at org.codehaus.enunciate.modules.FreemarkerDeploymentModule.processTemplate(FreemarkerDeploymentModule.java:106)
	at org.codehaus.enunciate.modules.FreemarkerDeploymentModule.processTemplate(FreemarkerDeploymentModule.java:85)
	at org.codehaus.enunciate.modules.FreemarkerDeploymentModule.processTemplate(FreemarkerDeploymentModule.java:70)
	at org.codehaus.enunciate.modules.xml.XMLDeploymentModule.doFreemarkerGenerate(XMLDeploymentModule.java:324)
	at org.codehaus.enunciate.modules.FreemarkerDeploymentModule.doGenerate(FreemarkerDeploymentModule.java:51)
	at org.codehaus.enunciate.modules.BasicDeploymentModule.step(BasicDeploymentModule.java:107)
	at org.codehaus.enunciate.apt.EnunciateAnnotationProcessor.process(EnunciateAnnotationProcessor.java:128)
	at com.sun.mirror.apt.AnnotationProcessors$CompositeAnnotationProcessor.process(AnnotationProcessors.java:84)
	at com.sun.tools.apt.comp.Apt.main(Apt.java:480)
	at com.sun.tools.apt.main.AptJavaCompiler.compile(AptJavaCompiler.java:270)
	at com.sun.tools.apt.main.Main.compile(Main.java:1127)
	at com.sun.tools.apt.main.Main.compile(Main.java:989)
	at com.sun.tools.apt.Main.processing(Main.java:113)
	at com.sun.tools.apt.Main.process(Main.java:103)
	at com.sun.tools.apt.Main.process(Main.java:85)
	at org.codehaus.enunciate.main.Enunciate.invokeApt(Enunciate.java:812)
	at org.codehaus.enunciate.main.Enunciate.doGenerate(Enunciate.java:396)
	at org.codehaus.enunciate.ConfigMojo$MavenSpecificEnunciate.doGenerate(ConfigMojo.java:646)
	at org.codehaus.enunciate.main.Enunciate$Stepper.step(Enunciate.java:1773)
	- locked <0x2a4dfec8> (a org.codehaus.enunciate.main.Enunciate$Stepper)
	at org.codehaus.enunciate.main.Enunciate$Stepper.stepTo(Enunciate.java:1805)
	- locked <0x2a4dfec8> (a org.codehaus.enunciate.main.Enunciate$Stepper)
	at org.codehaus.enunciate.DocsMojo.execute(DocsMojo.java:96)
	- locked <0x2a4dfed8> (a org.codehaus.enunciate.ThreadSafety)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)

"VM Thread" prio=10 tid=0x0036d400 nid=0x1a3c runnable 

"VM Periodic Task Thread" prio=10 tid=0x00405c00 nid=0x20c4 waiting on condition 

JNI global references: 150

**Comment by barkhakeni** [comment]
Wanted to follow up regarding this issue. Any ideas?

**Comment by stoicflame** [comment]
I can see the infinite loop, but it should have been broken at [SchemaInfo.java:293](https://github.com/stoicflame/enunciate/blob/master/core/src/main/java/org/codehaus/enunciate/config/SchemaInfo.java#L293).

Are you aware of any attempts to apply multiple threads to the build?

Can you reproduce it from the command line, without Eclipse?

In [Enunciate 2.0](https://github.com/stoicflame/enunciate/wiki/Enunciate-2), I've overhauled the state mechanism to not rely on `TheadLocal`s anymore... can you confirm the fix using 2.0.0-M.1?

**Comment by barkhakeni** [comment]
I tried using the command line and I see the same behavior. I will try with
enunciate 2.0. Could you please send me the link to it?

**Comment by stoicflame** [comment]
https://github.com/stoicflame/enunciate/wiki/Enunciate-2

**Comment by barkhakeni** [comment]
Thank you for the link. I used the latest enunciate plugin. I see the
following error. I am using Jackson 2.x. Not sure why i see this error.

[WARNING] [ENUNCIATE] Enunciate support for Jackson 1.x is deprecated. It
is recommended that you update to Jackson 2.x.
Jul 22, 2015 9:31:54 AM freemarker.log._JULLoggerFactory$JULLogger error
SEVERE: Error executing FreeMarker template
FreeMarker template error:
An error has occurred when reading existing sub-variable "body"; see cause
exception! The type of the containing value was: extended_hash+string
(com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl wrapped
into f.e.b.StringModel)

----
FTL stack trace ("~" means nesting-related):
- Failed at: ${type.example.body?xhtml}  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataType" at line 948, column 58]
~ Reached through: #nested  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "boilerplate" at line 118, column 9]
~ Reached through: @boilerplate title=(title + ": " + ty...  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataType" at line 842, column 5]
- Reached through: @file name=(type.slug + ".html")  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataType" at line 841, column 3]
- Reached through: @processDataType type=type  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataSyntax" at line 826, column 13]
~ Reached through: #nested  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "boilerplate" at line 118, column 9]
~ Reached through: @boilerplate title=(title + ": " + sy...  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataSyntax" at line 807, column 5]
- Reached through: @file name=(syntax.slug + ".html")  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataSyntax" at line 806, column 3]
- Reached through: @processDataSyntax syntax=syntax  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
at line 260, column 9]
~ Reached through: #nested  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "boilerplate" at line 118, column 9]
~ Reached through: @boilerplate pagenav=pagenav  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
at line 169, column 3]
- Reached through: @file name=indexPageName  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
at line 158, column 1]
----

Java stack trace (for programmers):
----
freemarker.core._TemplateModelException: [... Exception message was already
printed; see it above ...]
at freemarker.ext.beans.BeanModel.get(BeanModel.java:199)
at freemarker.core.Dot._eval(Dot.java:40)
at freemarker.core.Expression.eval(Expression.java:78)
at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
at freemarker.core.BuiltInForString._eval(BuiltInForString.java:26)
at freemarker.core.Expression.eval(Expression.java:78)
at freemarker.core.Expression.evalAndCoerceToString(Expression.java:82)
at freemarker.core.DollarVariable.accept(DollarVariable.java:40)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Environment.invokeNestedContent(Environment.java:535)
at freemarker.core.BodyInstruction.accept(BodyInstruction.java:56)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Macro$Context.runMacro(Macro.java:178)
at freemarker.core.Environment.invoke(Environment.java:700)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
at freemarker.core.Environment.visit(Environment.java:312)
at
freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
at
org.codehaus.enunciate.modules.docs.FileDirective.execute(FileDirective.java:48)
at freemarker.core.Environment.visit(Environment.java:377)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Macro$Context.runMacro(Macro.java:178)
at freemarker.core.Environment.invoke(Environment.java:700)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Environment.invokeNestedContent(Environment.java:535)
at freemarker.core.BodyInstruction.accept(BodyInstruction.java:56)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Macro$Context.runMacro(Macro.java:178)
at freemarker.core.Environment.invoke(Environment.java:700)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
at freemarker.core.Environment.visit(Environment.java:312)
at
freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
at
org.codehaus.enunciate.modules.docs.FileDirective.execute(FileDirective.java:48)
at freemarker.core.Environment.visit(Environment.java:377)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Macro$Context.runMacro(Macro.java:178)
at freemarker.core.Environment.invoke(Environment.java:700)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
at freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:159)
at freemarker.core.Environment.visitIteratorBlock(Environment.java:559)
at freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visitByHiddingParent(Environment.java:333)
at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:48)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Environment.invokeNestedContent(Environment.java:535)
at freemarker.core.BodyInstruction.accept(BodyInstruction.java:56)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Macro$Context.runMacro(Macro.java:178)
at freemarker.core.Environment.invoke(Environment.java:700)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at
freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
at
org.codehaus.enunciate.modules.docs.FileDirective.execute(FileDirective.java:48)
at freemarker.core.Environment.visit(Environment.java:377)
at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.MixedContent.accept(MixedContent.java:62)
at freemarker.core.Environment.visit(Environment.java:312)
at freemarker.core.Environment.process(Environment.java:290)
at freemarker.template.Template.process(Template.java:312)
at
org.codehaus.enunciate.modules.docs.DocumentationDeploymentModule.processTemplate(DocumentationDeploymentModule.java:391)
at
org.codehaus.enunciate.modules.docs.DocumentationDeploymentModule.call(DocumentationDeploymentModule.java:344)
at
com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
at
com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
at
rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
at
rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
at
rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
at
rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
at
rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
at
rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
at
rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
at
rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
at
rx.subjects.SubjectSubscriptionManager$SubjectObserver.onCompleted(SubjectSubscriptionManager.java:232)
at rx.subjects.ReplaySubject.onCompleted(ReplaySubject.java:404)
at rx.Observable$35.onCompleted(Observable.java:7458)
at rx.observers.SafeSubscriber.onCompleted(SafeSubscriber.java:81)
at
rx.internal.operators.OperatorDoOnEach$1.onCompleted(OperatorDoOnEach.java:53)
at
rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:110)
at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
at
rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
at
rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
at
rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
at
rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
at
rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
at
rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
at
rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
at
rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
at
rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
at
rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
at
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
at
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:606)
at freemarker.ext.beans.BeansWrapper.invokeMethod(BeansWrapper.java:1458)
at
freemarker.ext.beans.BeanModel.invokeThroughDescriptor(BeanModel.java:255)
at freemarker.ext.beans.BeanModel.get(BeanModel.java:159)
... 136 more
Caused by: com.webcohesion.enunciate.EnunciateException:
com.vce.vision.mvb.compliance.model.ComplianceProfileRule is not recognized
as a JSON type.
at
com.webcohesion.enunciate.modules.jackson1.model.types.JsonTypeVisitor.defaultAction(JsonTypeVisitor.java:45)
at
com.webcohesion.enunciate.modules.jackson1.model.types.JsonTypeVisitor.defaultAction(JsonTypeVisitor.java:41)
at
javax.lang.model.util.SimpleTypeVisitor6.visitDeclared(SimpleTypeVisitor6.java:157)
at
com.webcohesion.enunciate.modules.jackson1.model.types.JsonTypeVisitor.visitDeclared(JsonTypeVisitor.java:130)
at
com.webcohesion.enunciate.modules.jackson1.model.types.JsonTypeVisitor.visitDeclared(JsonTypeVisitor.java:41)
at
com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
at
com.webcohesion.enunciate.modules.jackson1.model.types.JsonTypeFactory.getJsonType(JsonTypeFactory.java:157)
at
com.webcohesion.enunciate.modules.jackson1.model.Accessor.getJsonType(Accessor.java:129)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.build(ExampleImpl.java:59)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.exampleNode(ExampleImpl.java:96)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.build(ExampleImpl.java:72)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.exampleNode(ExampleImpl.java:96)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.exampleNode(ExampleImpl.java:128)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.build(ExampleImpl.java:72)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.exampleNode(ExampleImpl.java:96)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.build(ExampleImpl.java:72)
at
com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl.getBody(ExampleImpl.java:42)
... 143 more

[INFO]
------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO]
------------------------------------------------------------------------
[INFO] Total time: 01:34 min
[INFO] Finished at: 2015-07-22T09:31:54-05:00
[INFO] Final Memory: 94M/237M
[INFO]
------------------------------------------------------------------------
[ERROR] Failed to execute goal
com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:docs (default)
on project mvb-compliance: Execution default of goal
com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:docs failed:
com.webcohesion.enunciate.EnunciateException:
freemarker.core._TemplateModelException: An error has occurred when reading
existing sub-variable "body"; see cause exception! The type of the
containing value was: extended_hash+string
(com.webcohesion.enunciate.modules.jackson1.api.impl.ExampleImpl wrapped
into f.e.b.StringModel)
[ERROR]
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] - Failed at: ${type.example.body?xhtml}  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataType" at line 948, column 58]
[ERROR] - Reached through: @file name=(type.slug + ".html")  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataType" at line 841, column 3]
[ERROR] - Reached through: @processDataType type=type  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataSyntax" at line 826, column 13]
[ERROR] - Reached through: @file name=(syntax.slug + ".html")  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
in macro "processDataSyntax" at line 806, column 3]
[ERROR] - Reached through: @processDataSyntax syntax=syntax  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
at line 260, column 9]
[ERROR] - Reached through: @file name=indexPageName  [in template
"jar:file:/C:/Users/kenib/.m2/repository/com/webcohesion/enunciate/enunciate-docs/2.0.0-M.1/enunciate-docs-2.0.0-M.1.jar!/org/codehaus/enunciate/modules/docs/docs.fmt"
at line 158, column 1]
[ERROR] (Hidden 6 "~" lines for terseness)
[ERROR]
[ERROR] ----: InvocationTargetException: test.model.ComplianceProfileRule
is not recognized as a JSON type. OnError while emitting onNext value:
com.webcohesion.enunciate.EnunciateContext.class
[ERROR] -> [Help 1]
[ERROR]


On Tue, Jul 21, 2015 at 7:15 PM, Ryan Heaton <notifications@github.com>
wrote:

> https://github.com/stoicflame/enunciate/wiki/Enunciate-2
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stoicflame/enunciate/issues/93#issuecomment-123505337>
> .
>



-- 
Thanks,
Barkha

**Comment by stoicflame** [comment]
What does your `com.vce.vision.mvb.compliance.model.ComplianceProfileRule` class look like? What do the references to that class look like?

**Comment by barkhakeni** [comment]
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "complianceProfileRule", propOrder = {"title", "description", "selected", "complianceProfileValues", "checked"})
public class ComplianceProfileRule
{

    protected String                  title;
    protected String                  description;
    @XmlElement(defaultValue = "false")
    protected Boolean                 selected;
    protected ComplianceProfileValues complianceProfileValues;
    @XmlAttribute(required = true)
    protected String                  id;
    @XmlElement(defaultValue = "false")
    protected Boolean                 checked = false;


It doesnt have the XMLRootElement

**Comment by barkhakeni** [comment]
When i do maven clean install, it doesnt work. But it works second time (just install). Also i have a docs goal but it doesnt create docs directory under target folder but i do see a docs directory under target/enunciate.

**Comment by stoicflame** [comment]
What classes _reference_ `ComplianceProfileRule`? I.e. which data types _use_ `ComplianceProfileRule` and what do their accessors look like? Is there something like this?

```java
protected List<ComplianceProfileRule> rules;
```

**Comment by barkhakeni** [comment]
Yes it is accessed as follows. Is there anything I am missing


@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "complianceProfileRules", propOrder =
{"complianceProfileRule"})
@XmlSeeAlso(value = {ComplianceProfileRule.class})
public class ComplianceProfileRules
{

    protected List<ComplianceProfileRule> complianceProfileRule;

On Wed, Jul 22, 2015 at 11:53 AM, Ryan Heaton <notifications@github.com>
wrote:

> What classes *reference* ComplianceProfileRule? I.e. which data types
> *use* ComplianceProfileRule and what do their accessors look like? Is
> there something like this?
>
> protected List<ComplianceProfileRule> rules;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stoicflame/enunciate/issues/93#issuecomment-123768732>
> .
>



-- 
Thanks,
Barkha

**Comment by stoicflame** [comment]
Okay. We'll get that error fixed in milestone 2.

I'm afraid I still have no way to debug the hung process in Enunciate 1. Is there some way you can provide me a way to reproduce the problem in my own environment?

**Comment by stoicflame** [comment]
Tracking the v2 error at #96. We'll keep this issue applied to v1.30.

**Comment by barkhakeni** [comment]
Thanks a lot Ryan. I will see if i can reproduce it with a small project. I am afraid it might be difficult but will try.
